lib/actions/deploy.js

```diff
@@ -296,20 +296,8 @@ Examples:
 					});
 				}
 			})
-			.then(() =>
-				Promise.join(
-					helpers.getApplication(appName),
```

Comment:
Previously we were calling `getApplication` here as well as in `getArchAndDeviceType`, by combining it all into `getAppWithArch` it allows us to save an api call, reduce duplicate code, and remove some `@ts-ignore`s